clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -82,6 +95,24 @@
 
     private final Time time;
 
+    private final Map<Integer, Collection<ApiVersionsResponse.ApiVersion>> nodeApiVersions = new HashMap<>();
+
+    private final Queue<String> apiVersionsRequestQueue = new ConcurrentLinkedQueue<>();
+
+    public NetworkClient(Selectable selector,
+                         Metadata metadata,
+                         String clientId,
+                         int maxInFlightRequestsPerConnection,
+                         long reconnectBackoffMs,
+                         int socketSendBuffer,
+                         int socketReceiveBuffer,
+                         int requestTimeoutMs,
+                         Time time,
+                         Collection<ApiVersionsResponse.ApiVersion> apis) {
```

Comment:
Hmm.  I was initially confused by what the "apis" parameter did.  Can we call this "requiredApiVersions"?

Same comment for the other constructor overloads.